e2e-tests/development-runtime/plugins/gatsby-source-fake-data/api.js

```diff
@@ -0,0 +1,103 @@
+const fs = require(`fs`)
+const util = require(`util`)
+const path = require(`path`)
+
+const NODE_TYPE = `FakeData`
+
+const readFile = file =>
+  util
+    .promisify(fs.readFile)(file, `utf8`)
+    .then(contents => JSON.parse(contents))
+    .catch(() => [])
+
+const writeFile = (file, contents) =>
+  util
+    .promisify(fs.writeFile)(file, JSON.stringify(contents), `utf8`)
+    .catch(() => null)
+
+const difference = (updated, existing) => {
+  const lookup = updated.reduce((merged, node) => {
+    merged[node.uuid] = node
+    return merged
+  }, {})
+  return existing.filter(node => !lookup[node.uuid])
+}
+
+const getNode = (data, { createNodeId, createContentDigest }) => {
+  const nodeContent = JSON.stringify(data)
+
+  const meta = {
+    id: createNodeId(`fake-data-${data.uuid}`),
+    parent: null,
+    children: [],
+    internal: {
+      type: NODE_TYPE,
+      content: nodeContent,
+      contentDigest: createContentDigest(data),
+    },
+  }
+
+  return Object.assign({}, data, meta)
+}
+
+const max = nodes =>
+  nodes.reduce((maxValue, node) => {
+    if (node.uuid > maxValue) {
+      return node.uuid
+    }
+    return maxValue
+  }, 0)
+
+module.exports = {
+  dbFilePath: path.join(__dirname, `db.json`),
+  nodeType: NODE_TYPE,
+  nodes: [],
+  async sync(helpers) {
+    // empty array
+    const nodes = await readFile(this.dbFilePath)
+
+    const uuid = max(nodes) + 1
```

Comment:
Nit: Could use `uuid`

Reply:
Eh - it functions as a uuid, don't see the need for an extra dep!